src/aws/flb_aws_credentials.c

```diff
@@ -535,32 +535,42 @@ time_t flb_aws_cred_expiration(const char *timestamp)
      return expiration;
 }
 
+size_t fsize(FILE *f)
+{
+    size_t len;
+    fseek(f, 0, SEEK_END);
+    len = (size_t)ftell(f);
+    rewind(f);
+    return len;
+}
+
 int flb_read_file(const char *path, char **out_buf, size_t *out_size)
 {
-    int ret;
     long bytes;
     char *buf = NULL;
     FILE *fp = NULL;
-    struct stat st;
+    size_t len;
 
-    ret = stat(path, &st);
-    if (ret == -1) {
+    fp = fopen(path, "r");
+    if (!fp) {
         return -1;
     }
 
-    fp = fopen(path, "r");
-    if (!fp) {
+    len = fsize(fp);
```

Comment:
that function is not necessary.

Since you already have an open file through ```fp```, you can get it file descriptor and use fstat(2) to get the file size, e.g:

```c
    int ret;
    struct stat st;

    fd = fileno(fp);
    ret = fstat(fd, &st);
    if (ret == -1) {
        flb_errno();
        ....
    }

    /* your file size is on st.st_size */
```

Reply:
that mechanism is safe